test/e2e/identityvalidator/identityvalidator.go

```diff
@@ -0,0 +1,152 @@
+	if err != nil || t2 == nil {
+		logger.Fatalf("testMSIEndpointFromUserAssignedID failed, %+v", err)
+	}
+
+	// Check if the above two tokens are the same
+	if !strings.EqualFold(t1.AccessToken, t2.AccessToken) {
+		logger.Fatalf("msi, emsi test failed %+v %+v", t1, t2)
+	}
+
+	// Test 4
+	if err := testInstanceMetadataRequests(logger); err != nil {
+		logger.Fatalf("testInstanceMetadataRequests failed, %+v", err)
+	}
+}
+
+func doARMOperations(logger *log.Entry, subscriptionID, resourceGroup string) error {
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		logger.Errorf("failed NewAuthorizerFromEnvironment  %+v", authorizer)
+		return err
+	}
+	vmClient := compute.NewVirtualMachinesClient(subscriptionID)
+	vmClient.Authorizer = authorizer
+	vmlist, err := vmClient.List(context.Background(), resourceGroup)
+	if err != nil {
+		logger.Errorf("failed to list all vm %+v", err)
+		return err
+	}
+
+	logger.Infof("succesfull doARMOperations vm count %d", len(vmlist.Values()))
+	return nil
+}
+
+func testMSIEndpoint(logger *log.Entry, msiEndpoint, resource string) (*adal.Token, error) {
+	spt, err := adal.NewServicePrincipalTokenFromMSI(msiEndpoint, resource)
+	if err != nil {
+		logger.Errorf("failed to acquire a token using the MSI VM extension, Error: %+v", err)
+		return nil, err
+	}
+	if err := spt.Refresh(); err != nil {
+		logger.Errorf("failed to refresh ServicePrincipalTokenFromMSI using the MSI VM extension, msiEndpoint(%s)", msiEndpoint)
+		return nil, err
+	}
+	token := spt.Token()
+	if token.IsZero() {
+		logger.Errorf("zero token found, MSI VM extension, msiEndpoint(%s)", msiEndpoint)
+		return nil, err
+	}
+	logger.Infof("succesfully acquired a token using the MSI, msiEndpoint(%s)", msiEndpoint)
+	return &token, nil
+}
+
+func testMSIEndpointFromUserAssignedID(logger *log.Entry, msiEndpoint, userAssignedID, resource string) (*adal.Token, error) {
+	spt, err := adal.NewServicePrincipalTokenFromMSIWithUserAssignedID(msiEndpoint, resource, userAssignedID)
+	if err != nil {
+		logger.Errorf("failed NewServicePrincipalTokenFromMSIWithUserAssignedID, clientID: %s Error: %+v", userAssignedID, err)
+		return nil, err
+	}
+	if err := spt.Refresh(); err != nil {
+		logger.Errorf("failed to refresh ServicePrincipalToken userAssignedID MSI, msiEndpoint(%s)", msiEndpoint)
+		return nil, err
+	}
+	token := spt.Token()
+	if token.IsZero() {
+		logger.Errorf("zero token found, userAssignedID MSI, msiEndpoint(%s) clientID(%s)", msiEndpoint, userAssignedID)
+		return nil, err
+	}
+	logger.Infof("succesfully acquired a token, userAssignedID MSI, msiEndpoint(%s) clientID(%s)", msiEndpoint, userAssignedID)
+	return &token, err
+}
+
+// Simulates health probe of non MSI instance metadata requests
+// e.g. curl -H Metadata:true "http://169.254.169.254/metadata/instance?api-version=2017-08-01" --header "X-Forwarded-For: 192.168.0.2"
+func testInstanceMetadataRequests(logger *log.Entry) error {
```

Comment:
Is there a reason for us to try this, if the tests above passes ?

Reply:
Right, I will delete this test function